internal/cloudinit/cloudinit.go

```diff
@@ -395,7 +399,16 @@ func LinuxBash(params *Params) (payload string) {
 	}
 
 	var err error
-	if params.GitspaceAgentConfig.Secret != "" && params.GitspaceAgentConfig.AccessToken != "" {
+	if (params.GitspaceAgentConfig.Secret != "" && params.GitspaceAgentConfig.AccessToken != "") ||
+		(params.GitspaceAgentConfig.VMInitScript != "") {
+		if params.GitspaceAgentConfig.VMInitScript != "" {
+			decodedScript, decodeErr := base64.StdEncoding.DecodeString(params.GitspaceAgentConfig.VMInitScript)
+			if decodeErr != nil {
+				err = fmt.Errorf("failed to decode the gitspaces vm init script: %w", err)
+				panic(err)
```

Comment:
Should we panic here? How about we return the error?

Reply:
That's the convention followed throughout this file, so I continued doing so 🙂

Reply:
I agree that we shouldn't panic (even though I believe we handle stacktraces and don't crash the program but I may be wrong). Can you change this (and other instances of panic if possible otherwise we can take that up later). The error can be propagated up

Reply:
Sure I can do it. It would mean changes in a lot of files. I would prefer to take it separately in a different PR

Reply:
Around 14 files